File: auth_service/methods/mailer.js
const nodemailer = require('nodemailer');

const EMAIL_USER = process.env.EMAIL_USER;
const EMAIL_PASS = process.env.EMAIL_PASS;

if(!EMAIL_USER || !EMAIL_PASS){
    console.error('FATAL ERROR: EMAIL_USER or EMAIL_PASS is not defined.');
    process.exit(1);
} else {
    console.log('EMAIL_USER and EMAIL_PASS are defined');
}

const transporter = nodemailer.createTransport({
    host: 'smtp.gmail.com',
    port: 587,  // Use 465 for SSL, or 587 for TLS
    auth: {
        user: EMAIL_USER,
        pass: EMAIL_PASS,
    },
});

// Function to send email
const sendEmail = async (to, subject, body) => {
    try{
        // const testResults = await transporter.verify();
        // console.log('Test email results:', testResults);
        await transporter.sendMail({
            from: EMAIL_USER,
            to: to,
            subject: subject,
            html: body,
        });
        console.log(`Email sent to ${to} with subject: ${subject}`);
    } catch(error){
        console.error("Error sending email:", error);
        return error;
    }
}


module.exports = {
    sendEmail
};

File: notification_system/dispatch/utils/mailer.py
import os
import smtplib
from email.mime.text import MIMEText
from email.mime.multipart import MIMEMultipart

EMAIL_USER = os.getenv("EMAIL_USER")
EMAIL_PASS = os.getenv("EMAIL_PASS")

if not EMAIL_USER or not EMAIL_PASS:
    print("FATAL ERROR: EMAIL_USER or EMAIL_PASS is not defined.")
    exit(1)
else:
    print("EMAIL_USER and EMAIL_PASS are defined")

def send_email_to_recipient(to, subject, body):
    try:
        msg = MIMEMultipart()
        msg["From"] = EMAIL_USER
        msg["To"] = to
        msg["Subject"] = subject

        msg.attach(MIMEText(body, "html"))

        with smtplib.SMTP("smtp.gmail.com", 587) as server:
            server.starttls()  
            server.login(EMAIL_USER, EMAIL_PASS)
            server.sendmail(EMAIL_USER, to, msg.as_string())

        print(f"Email sent to {to} with subject: {subject}")
    except Exception as error:
        print(f"Error sending email: {error}")
        return error